### PR TITLE
Update truffle-migrate and truffle-require

### PR DESCRIPTION
**Pull request by eggplantzzz**
Update Require.file to no longer take a callback, use synchronous version of fs method, and update some syntax. Also remove callbacks from truffle-migrate.

### CODE REVIEW

**Comment by CplXhristianUSMC** [comment]
Thank you so much for your reply I'm sorry that it's taking me so long to
get back to your emails currently I'm in the process of moving and will be
driving no less than 1300 miles tonight across state lines to get to my new
house please forgive any late replies I greatly appreciate any and all work
that you have been doing to help rectify this matter truly thankful and
gratefully yours cplxhristianusmc

**Comment by eggplantzzz** [comment]
Hmmm, looks like those are also failing on `develop`. I guess we'll have to figure out how to resolve those.

File: packages/truffle-core/lib/commands/test.js [hunk 1]
```diff
@@ -123,26 +121,29 @@ var command = {
...
+            .catch(error => {
+              done(error);
             });
-          });
         };
 
         if (config.networks[config.network]) {
           Environment.detect(config)
             .then(() => {
+              console.log("IN the then from detect!!!!!!");
```
Comment (CruzMolina) [inline packages/truffle-core/lib/commands/test.js]:
```suggestion
```

File: packages/truffle-core/lib/commands/test.js [hunk 2]
```diff
@@ -162,7 +163,7 @@ var command = {
             gasLimit: config.gas,
             noVMErrorsOnRPCResponse: true
           };
-
+          console.log("about to run Develop.connectOrStart");
```
Comment (CruzMolina) [inline packages/truffle-core/lib/commands/test.js]:
```suggestion
```

**Pull request merged**